shared/src/vyxal/parsing/Lexer.scala

```diff
@@ -150,8 +160,88 @@ object Lexer:
   def lexSBCS(code: String): Either[VyxalCompilationError, List[Token]] =
     SBCSLexer.lex(code)
 
+  def performMoves(tkns: List[LitToken]): List[LitToken] =
+    val tokens = tkns.map {
+      case LitToken(TokenType.Group, tokens, range) => LitToken(
+          TokenType.Group,
+          performMoves(tokens.asInstanceOf[List[LitToken]]),
+          range,
+        )
+      case token => token
+    }
+    val merged = ListBuffer[LitToken]()
+    for token <- tokens do
+      if token.tokenType == TokenType.MoveRight then
+        if merged.nonEmpty && merged.last.tokenType == TokenType.MoveRight then
+          merged.last.copy(value =
+            merged.last.value.asInstanceOf[String] +
+              token.value.asInstanceOf[String]
+          )
+        else merged += token
+      else merged += token
+
+    // Now, bind the move right tokens to the next token
+
+    var bound = ListBuffer[LitToken | Tuple2[LitToken, Int]]()
```

Comment:
```suggestion
    var bound = ListBuffer[LitToken | (LitToken, Int)]()
```
You used Copilot again, didn't you?

Reply:
Nope, this was me. I liked the explicit tuple2